src/modules/dashboard/components/Extensions/ExtensionSetup.tsx

```diff
@@ -178,6 +237,107 @@ const ExtensionSetup = ({
 
   if (!initializationParams) return null;
 
+  const displayParams = (params, values, isExtraParams) =>
+    params.map(
+      ({
+        paramName,
+        title,
+        fieldName,
+        description,
+        type,
+        options,
+        disabled,
+        complementaryLabel,
+        tokenLabel,
+      }) => (
+        <div
+          key={paramName}
+          className={isExtraParams ? styles.extraParams : ''}
+        >
+          {type === ExtensionParamType.Input && (
+            <div className={styles.input}>
+              <Input
+                appearance={{ size: 'medium', theme: 'minimal' }}
+                label={title}
+                name={paramName}
+              />
+              <FormattedMessage
+                {...description}
+                values={{
+                  span: (chunks) => (
+                    <span className={styles.descriptionExample}>{chunks}</span>
+                  ),
+                }}
+              />
+              {(complementaryLabel || tokenLabel) && (
+                <span className={styles.complementaryLabel}>
+                  {complementaryLabel ? (
+                    <FormattedMessage {...MSG[complementaryLabel]} />
+                  ) : (
+                    getToken(values[tokenLabel])?.symbol
+                  )}
+                </span>
+              )}
+            </div>
+          )}
+          {type === ExtensionParamType.Textarea && (
+            <div className={styles.textArea}>
+              <Textarea
+                appearance={{ colorSchema: 'grey' }}
+                label={title}
+                name={paramName}
+                disabled={disabled && disabled(values)}
+              />
+              {description && (
+                <p className={styles.textAreaDescription}>
+                  <FormattedMessage {...description} />
+                </p>
+              )}
+            </div>
+          )}
+          {type === ExtensionParamType.ColonyPolicySelector && (
+            <ColonyPolicySelector
+              name={paramName}
+              title={title}
+              options={options || []}
+            />
+          )}
+          {type === ExtensionParamType.TokenSelector && (
+            <div>
+              <InputLabel
+                label={title}
+                appearance={{
+                  theme: 'minimal',
+                  size: 'medium',
+                }}
+              />
+              <div className={styles.tokenSelectorContainer}>
+                {fieldName && (
+                  <p>
+                    <FormattedMessage {...fieldName} />
+                  </p>
+                )}
+                <TokenSymbolSelector
+                  tokens={tokens}
+                  name={paramName}
+                  appearance={{ alignOptions: 'right', theme: 'grey' }}
+                  elementOnly
+                  label={paramName}
+                />
```

Comment:
There's a bug in the `TokenSymbolSelector` component. Either only here, or is a global problem.

When changing the token, the token's avatar doesn't change.

Here's a demo of the bug:
![bug-token-selector-same-token-avatar](https://user-images.githubusercontent.com/1193222/131263513-5c462f51-fdaa-4483-8c1b-06954e645660.gif)

Reply:
I'll check again, maybe it was something on my side

Reply:
Re-tested and it's the same :(

![bug-token-selector-same-token-avatar-2](https://user-images.githubusercontent.com/1193222/131343513-652a997c-d352-4661-aa02-8424d6c3fb52.gif)
![bug-token-selector-same-token-avatar-3](https://user-images.githubusercontent.com/1193222/131343517-0e285959-0ebe-4a9c-9241-be51a9fd6dde.gif)
![bug-token-selector-same-token-avatar-4](https://user-images.githubusercontent.com/1193222/131343518-d1c340eb-9a03-4803-9250-ac1218ee15ff.gif)

Tell you what. I'll approve this PR, since this bug is not really related to this issue, but after you merge this, open a issue to track the Token Selector bug and fix it separately